src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyDiscoveryManager.cs

```diff
@@ -134,63 +148,44 @@ public bool HandlePartialDiscoveryComplete(IProxyDiscoveryManager proxyDiscovery
         */
         if (allDiscoverersCompleted || IsAbortRequested)
         {
-            // Reset enumerators
-            _sourceEnumerator = null;
-
-            _currentDiscoveryDataAggregator = null;
-            _currentDiscoveryEventsHandler = null;
-
-            // Dispose concurrent executors
-            UpdateParallelLevel(0);
+            _parallelOperationManager.StopAllManagers();
 
             return true;
         }
 
-        /*  Discovery is not complete.
```

Comment:
This comment is wrong, up until now .NET and .NET Framework would not run in the same run. When it was added it also changed the logic and stopped respecting Shared hosts.